include/libdnf/conf/option_bool.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_CONF_OPTION_BOOL_HPP
+#define LIBDNF_CONF_OPTION_BOOL_HPP
+
+
+#include "option.hpp"
+
+#include <array>
+#include <memory>
+#include <vector>
+
+namespace libdnf {
+
+
+class OptionBool : public Option {
+public:
+    using ValueType = bool;
+
+    class InvalidValue : public Option::InvalidValue {
+    public:
+        using Option::InvalidValue::InvalidValue;
+        const char * get_domain_name() const noexcept override { return "libdnf::OptionBool"; }
+    };
+
+    explicit OptionBool(bool default_value);
+    OptionBool(
+        bool default_value, const std::vector<std::string> & true_vals, const std::vector<std::string> & false_vals);
+    OptionBool(const OptionBool & src);
+    OptionBool(OptionBool && src) noexcept = default;
+    ~OptionBool() override = default;
+
+    OptionBool & operator=(const OptionBool & src);
+    OptionBool & operator=(OptionBool && src) noexcept = default;
+
+    OptionBool * clone() const override;
+    void test(bool /*unused*/) const;
+    bool from_string(const std::string & value) const;
+    void set(Priority priority, bool value);
+    void set(Priority priority, const std::string & value) override;
+    bool get_value() const noexcept;
+    bool get_default_value() const noexcept;
+    std::string to_string(bool value) const;
+    std::string get_value_string() const override;
+    static const std::vector<std::string> & get_default_true_values() noexcept;
+    static const std::vector<std::string> & get_default_false_values() noexcept;
+    const std::vector<std::string> & get_true_values() const noexcept;
+    const std::vector<std::string> & get_false_values() const noexcept;
```

Comment:
Four above function looks cryptic to me. Somehow I don't now or see a significant difference. Please could you provide additional details?

Reply:
The `get_true_values()` and `get_false_values()` return vectors that contains actual values used for conversion to/from string.
But get_default_true_values() and get_default_false_values() return vectors with "defaults". This "defaults" are used until we replace it by our values - parameter of constructor.

In old libdnf tha vectors returned by get_default_*_values() were a global constants. But they are encapsulated in the bool class method now. Theoreticaly I can make they protected. We don't using they now.